### PR TITLE
Consume miner actor refactor

### PR DESCRIPTION
**Pull request by magik6k**
Blocked on https://github.com/filecoin-project/specs-actors/pull/648. This PR should have the bulk of changes which we'll need to consume the actors PR

Depends on:
- [x] https://github.com/filecoin-project/specs-actors/pull/648
- [x] https://github.com/filecoin-project/storage-fsm/pull/48
- [x] chain-val

### CODE REVIEW

File: api/api_full.go [hunk 1]
```diff
@@ -267,7 +269,13 @@ type FullNode interface {
 	// StateSectorPreCommitInfo returns the PreCommit info for the specified miner's sector
 	StateSectorPreCommitInfo(context.Context, address.Address, abi.SectorNumber, types.TipSetKey) (miner.SectorPreCommitOnChainInfo, error)
 	// StateSectorGetInfo returns the on-chain info for the specified miner's sector
+	// NOTE: returned info.Expiration may not be accurate in some cases, use StateSectorExpiration to get accurate
+	// expiration epoch
 	StateSectorGetInfo(context.Context, address.Address, abi.SectorNumber, types.TipSetKey) (*miner.SectorOnChainInfo, error)
+	// StateSectorExpiration returns epoch at which given sector will expire
+	StateSectorExpiration(context.Context, address.Address, abi.SectorNumber, types.TipSetKey) (*SectorExpiration, error)
+	// StateSectorPartition finds deadline/partition with the specified sector
+	StateSectorPartition(ctx context.Context, maddr address.Address, sectorNumber abi.SectorNumber, tok types.TipSetKey) (*SectorLocation, error)
```
Comment (arajasek) [inline api/api_full.go:278]: Wanna remove the param names?

File: api/api_full.go [hunk 2]
```diff
@@ -358,6 +366,19 @@ type MinerSectors struct {
 	Pset uint64
 }
 
+type SectorExpiration struct {
+	OnTime abi.ChainEpoch
+
+	// non-zero if sector is faulty, epoch at which it will it will be
```
Comment (arajasek) [inline api/api_full.go:372]: "it will" twice in the comment

**Comment by arajasek** [comment]
oh, needs an API version bumpy bump

File: node/impl/full/state.go [hunk 3]
```diff
@@ -65,19 +70,42 @@ func (a *StateAPI) StateMinerSectors(ctx context.Context, addr address.Address,
...
-	var mas miner.State
-	_, err = a.StateManager.LoadActorState(ctx, addr, &mas, ts)
+	err := a.StateManager.WithParentStateTsk(tsk,
+		a.StateManager.WithActor(maddr,
+			a.StateManager.WithActorState(ctx, func(store adt.Store, mas *miner.State) error {
+				var allActive []*abi.BitField
+
+				err := a.StateManager.WithDeadlines(
+					a.StateManager.WithEachDeadline(
+						a.StateManager.WithEachPartition(func(store adt.Store, partIdx uint64, partition *miner.Partition) error {
```
Comment (Kubuxu) [inline node/impl/full/state.go:83]: IMO this is more complex than it has to be but meh.

File: api/test/deals.go [hunk 4]
```diff
@@ -191,14 +191,17 @@ func startSealingWaiting(t *testing.T, ctx context.Context, miner TestStorageNod
 	snums, err := miner.SectorsList(ctx)
 	require.NoError(t, err)
 
+	fmt.Printf("Miner sector states: ")
```
Comment (Kubuxu) [inline api/test/deals.go:194]: use `t.Logf` in tests

File: api/api_full.go (hunk 1 above)
Comment (magik6k) [reply api/api_full.go:278]: IMO we should add them to other methods, not remove

(And then actually document what they do)

**Pull request merged**